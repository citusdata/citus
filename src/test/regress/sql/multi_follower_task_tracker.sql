\c - - - :master_port

-- do some setup

CREATE TABLE tab(a int, b int);
SELECT create_distributed_table('tab', 'a');

INSERT INTO tab (a, b) VALUES (1, 1);
INSERT INTO tab (a, b) VALUES (1, 2);

\c - - - :follower_master_port

SET citus.task_executor_type TO 'real-time';
SELECT * FROM tab;
SET citus.task_executor_type TO 'task-tracker';
SELECT * FROM tab;

-- clean up

\c - - - :master_port

DROP TABLE tab;
