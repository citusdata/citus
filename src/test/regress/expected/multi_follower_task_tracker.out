\c - - - :master_port
-- do some setup
CREATE TABLE tab(a int, b int);
SELECT create_distributed_table('tab', 'a');
 create_distributed_table 
--------------------------
 
(1 row)

INSERT INTO tab (a, b) VALUES (1, 1);
INSERT INTO tab (a, b) VALUES (1, 2);
\c - - - :follower_master_port
RESET citus.task_executor_type;
SELECT * FROM tab;
 a | b 
---+---
 1 | 1
 1 | 2
(2 rows)

SET citus.task_executor_type TO 'task-tracker';
SELECT * FROM tab;
ERROR:  task tracker queries are not allowed while citus.use_secondary_nodes is 'always'
HINT:  try setting citus.task_executor_type TO 'adaptive'
-- clean up
\c - - - :master_port
DROP TABLE tab;
